### DISCUSSION TITLE
Cross compilation for armv7

### DISCUSSION THREAD

**Discussion by schnapper79** [slint-ui/slint#2671] [Q&A] [answered]
I am aware about the question regarding cross compilation for pi 2/3/4 using cross.

I have a Debian host with and an arm sysroot and a cross toolchain. What libraries I need to cross compile a rust slint project using winit-femtovg? And how can I tell rust to use the sysroot when looking for these libraries?

**Comment by tronical** [comment, marked as answer]
At minimum in your target sysroot you'll need fontconfig, either xcb or wayland, and libxkbcommon. We do something like this in our docker images at https://github.com/slint-ui/slint/blob/master/docker/Dockerfile.armv7-unknown-linux-gnueabihf - those install a bit more though to also cover Skia (fontconfig-dev packages and libstdc++ for target, and clang/python3 for the host).

**Comment by schnapper79** [comment]
But how do I tell cargo (?) where my sysroot resides?

Simon Hausmann ***@***.***> schrieb am Di., 2. Mai 2023,
09:06:

**Reply by tronical** [reply]
I believe you won't need to as the pure Rust build won't look for any libraries to link against. Wayland/xcb/fontconfig are opened dynamically at runtime using dlopen.

**Comment by schnapper79** [comment]
Ah great thats perfect

Simon Hausmann ***@***.***> schrieb am Di., 2. Mai 2023,
09:36:
